lib/srv/regular/sshserver.go

```diff
@@ -1013,6 +1025,70 @@ func (s *Server) getServerResource() (types.Resource, error) {
 	return s.getServerInfo(), nil
 }
 
+func (s *Server) filterExistingNodes(instances *server.EC2Instances) (*server.EC2Instances, error) {
```

Comment:
can we keep a cached version of this data somehow?
For large ec2 fleets, `GetNodes` might extract a lot of info (including nodes that weren't discovered by ec2 discovery) from AuthServer and we can cache everything locally.

One way:
On load, the agent loads from Auth server the list of discovered servers (filter by tag so we do not receive other nodes) and stores them locally. Each time a new machine is discovered(not in cache) you can just ping AuthServer to see if no one else discovered that machine already and you update your local cache. If the machine was not discovered yet, you just run SSM against it

Reply:
I talked to @lxea about this today, we already use AccessPoint here which uses local cache rather than calling directly Auth server so I think we should be good.